app-android/src/main/kotlin/social/androiddev/dodo/MainActivity.kt

```diff
@@ -25,7 +26,10 @@ class MainActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
 
         // Create the root component before starting Compose
-        val root = DefaultRootComponent(componentContext = defaultComponentContext())
+        val root = DefaultRootComponent(
+            componentContext = defaultComponentContext(),
+            mainContext = Dispatchers.Main,
```

Comment:
nitty nit: `Dispatchers.Main.immediate`

Reply:
I mimicked the default dispatcher for `viewModeScope`, but I don't see an issue with switching to `immediate`.